clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp

```diff
@@ -4453,7 +4453,13 @@ CIRGenFunction::emitAArch64BuiltinExpr(unsigned BuiltinID, const CallExpr *E,
   }
   case NEON::BI__builtin_neon_vldap1_lane_s64:
   case NEON::BI__builtin_neon_vldap1q_lane_s64: {
-    llvm_unreachable("NEON::BI__builtin_neon_vldap1q_lane_s64 NYI");
+    Ops[1] = builder.createBitcast(Ops[1], vTy);
+    cir::LoadOp Load = builder.createAlignedLoad(
+        Ops[0].getLoc(), vTy.getEltType(), Ops[0], PtrOp0.getAlignment());
+    Load.setAtomic(cir::MemOrder::Acquire);
+    Ops[0] = Load;
```

Comment:
You probably don't need to update `Ops[1]` and `Ops[0]` here, just pass the Load and do the bitcast directly while creating the insert op.